litellm/main.py

```diff
@@ -2176,13 +2176,23 @@ def completion(
             # boto3 reads keys from .env
             custom_prompt_dict = custom_prompt_dict or litellm.custom_prompt_dict
 
-            if (
-                "aws_bedrock_client" in optional_params
-            ):  # use old bedrock flow for aws_bedrock_client users.
-                response = bedrock.completion(
+                
+            if ("aws_bedrock_client" in optional_params):  
+                # Extract credentials for legacy boto3 client and pass thru to httpx
+                aws_bedrock_client = optional_params.pop("aws_bedrock_client")
+                creds = aws_bedrock_client._get_credentials().get_frozen_credentials()
+                if creds.access_key:
+                    optional_params["aws_access_key_id"] = creds.access_key
+                if creds.secret_key:
+                    optional_params["aws_secret_access_key"] = creds.secret_key
+                if creds.token:
+                    optional_params["aws_session_token"] = creds.token  
```

Comment:
don't these expire?

Reply:
Yes, they do. That is the nature of STS tokens. It limits exposure of AWS credentials by team members so that they can't save credentials to a config file and expose security to other people.  They have to always go back to the STS service to get new ones when ever they start new work.  Expiration can be set anywhere  up to, I think, 12 hours.

Reply:
Reference: 
https://docs.aws.amazon.com/whitepapers/latest/navigating-gdpr-compliance/temporary-access-tokens-through-aws-sts.html#:~:text=After%20temporary%20credentials%20expire%2C%20AWS,security%20credentials%20with%20an%20application.

> - Temporary security credentials are for short-term use. You can configure the amount of time that they are valid, from 15 minutes up to a maximum of 12 hours. After temporary credentials expire, AWS does not recognize them or allow any kind of access from API requests made with them.
> 
> - Temporary security credentials are not stored with the user. Instead, they are generated dynamically and provided to the user when requested. When (or before) temporary security credentials expire, a user can request new credentials, if that user has permissions to do so.
> 
> These differences provide the following advantages when you use temporary credentials:
> 
> - You do not have to distribute or embed long-term AWS security credentials with an application.
> 
> - Temporary credentials are the basis for roles and identity federation. You can provide access to your AWS resources to users by defining a temporary AWS identity for them.
> 
> - Temporary security credentials have a limited customizable lifespan. Because of this, you do not have to rotate them or explicitly revoke them when they're no longer needed. After temporary security credentials expire, they cannot be reused. You can specify the maximum amount of time the credentials are valid.